src/wp-includes/block-template-utils.php

```diff
@@ -1430,6 +1430,17 @@ function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '
 		$template_hierarchy[] = 'singular';
 	}
 	$template_hierarchy[] = 'index';
+
+	$template_type = '';
+	if ( ! empty( $template_prefix ) ) {
+		list( $template_type ) = explode( '-', $template_prefix );
+	} else {
+		list( $template_type ) = explode( '-', $slug );
+	}
+	$valid_template_types = array( '404', 'archive', 'attachment', 'author', 'category', 'date', 'embed', 'frontpage', 'home', 'index', 'page', 'paged', 'privacypolicy', 'search', 'single', 'singular', 'tag', 'taxonomy' );
+	if ( in_array( $template_type, $valid_template_types ) ) {
+		return apply_filters( "{$template_type}_template_hierarchy", $template_hierarchy );
```

Comment:
The document is missing for filter.

Reply:
Thanks for letting me know, I just added it in e742973803309d12edb0bf4a1475bf8a31ec3580. Given that this filter is documented in `wp-includes/template.php`, my understanding is that adding a comment like `This filter is documented in wp-includes/template.php` is the correct approach? I took these two examples ([1](https://github.com/WordPress/wordpress-develop/blob/b094b53a26600ca4cd6329919f5b3c3f55613c76/src/wp-admin/my-sites.php#L79-L80), [2](https://github.com/WordPress/wordpress-develop/blob/b094b53a26600ca4cd6329919f5b3c3f55613c76/src/wp-links-opml.php#L66C2-L67)) as a reference.